js/src/tooltip.js

```diff
@@ -589,6 +589,10 @@ class Tooltip {
         this.element.getAttribute('title') || ''
       )
 
+      if (!this.element.getAttribute('aria-label') && !this.element.textContent) {
```

Comment:
might be worth checking `&& this.element.getAttribute('title')` as well, as otherwise it'll add an unnecessary `aria-label` to situations where it's completely borked anyway (no text content, no `title`)

Reply:
this can still lead to a few edge cases where `aria-label` potentially overrides something inside the element that is not directly `textContent` - for instance, if it contained `<svg aria-label="foo">...</svg>` (which won't be exposed as `this.element.textContent` I believe) but where the element has a `title`. in that case, the `aria-label` will be given the content of the `title`, and anything else inside the element will not be announced...but this may be an unusual situation, particularly when the author already wants to use a tooltip on a control...